### ISSUE TITLE
"No pg_hba.conf entry" when adding Heroku databases from Mac app

### ISSUE THREAD

**Issue by crsimons1991** [metabase/metabase#1293]
pg_hba.conf entry for host "***********", user "***********", database "***********", ssl off.

**Comment by tbeseda** [comment]
Same issue here. Trying to connect to a PostgreSQL instance on Heroku (SSL required).
Maybe it's not defaulting to SSL connection?

**Comment by tlrobinson** [comment]
So it looks like this is only broken in the initial setup screen. Try skipping adding the database there, then once you're in the main app go to /admin/databases/create to add the database.

Let me know if that helps. Sorry about that!

**Comment by tbeseda** [comment]
@tlrobinson thanks for the quick reply.
I get the same error from the Admin, though:
![metabase-error](https://cloud.githubusercontent.com/assets/15697/10644467/1b27792c-77e6-11e5-8f7b-bac08969c829.png)

Note: I'm using the Mac client.

**Comment by crsimons1991** [comment]
I had the same issue after skipping the initial setup screen and going into
the admin section as well.

**Comment by tlrobinson** [comment]
Ok, I've verified it's a problem on both screens in the Mac app.

This kind of defeats the purpose of the Mac app, but a possible temporary workaround (if you're comfortable with the command line) would be to quit Metabase.app, then in Terminal.app run the following:

```
cd ~/Library/Application\ Support/Metabase
java -jar /Applications/Metabase.app/Contents/Resources/metabase.jar
```

(Assuming you moved Metabase.app to the Applications folder)

This will start Metabase manually with the same db the Mac app uses. The open up http://localhost:3000/ log in, and add the database at http://localhost:3000/admin/databases/create

Sorry about that. We'll get it fixed asap.

**Comment by dpaola2** [comment]
FWIW @tlrobinson when attempting your solution, both chrome and safari hang when hitting http://localhost:3000.

**Comment by dpaola2** [comment]
The server is definitely running, here's the output:

```
➜  Metabase  java -jar /Applications/Metabase.app/Contents/Resources/metabase.jar
2015-10-21 11:46:09,916 [INFO ] org.eclipse.jetty.util.log :: Logging initialized @1683ms
2015-10-21 11:46:15,697 [INFO ] metabase.core :: Starting Metabase in STANDALONE mode
2015-10-21 11:46:15,704 [INFO ] metabase.core :: Starting Metabase version v0.12.0-test (84d0afc master)...
2015-10-21 11:46:15,705 [INFO ] metabase.db :: Verifying Database Connection ...
2015-10-21 11:46:16,333 [INFO ] metabase.db :: Verify Database Connection ... CHECK
2015-10-21 11:46:19,001 [INFO ] metabase.db :: Database Migrations Current ... CHECK
2015-10-21 11:46:19,213 [INFO ] metabase.db.migrations :: Data migration: Setting database/table/type fields on all Cards.
2015-10-21 11:46:19,218 [INFO ] metabase.events :: Starting events system ...
2015-10-21 11:46:19,270 [DEBUG] metabase.db.internal :: DB CALL:  Card [:id :dataset_query] (select* ENTITY22161)
2015-10-21 11:46:19,485 [INFO ] metabase.events :: 	loading events namespace:  metabase.events.activity-feed
2015-10-21 11:46:19,514 [INFO ] metabase.events.activity-feed :: Starting activity-feed events listener
2015-10-21 11:46:19,524 [INFO ] metabase.events :: 	loading events namespace:  metabase.events.last-login
2015-10-21 11:46:19,531 [INFO ] metabase.events.last-login :: Starting last-login events listener
2015-10-21 11:46:19,533 [INFO ] metabase.events :: 	loading events namespace:  metabase.events.revision
2015-10-21 11:46:19,541 [INFO ] metabase.events.revision :: Starting revision events listener
2015-10-21 11:46:19,543 [INFO ] metabase.events :: 	loading events namespace:  metabase.events.sync-database
2015-10-21 11:46:19,551 [INFO ] metabase.events.sync-database :: Starting database sync events listener
2015-10-21 11:46:19,552 [INFO ] metabase.events :: 	loading events namespace:  metabase.events.view-log
2015-10-21 11:46:19,560 [INFO ] metabase.events.view-log :: Starting view-log events listener
2015-10-21 11:46:20,012 [DEBUG] metabase.task :: Starting Quartz Scheduler
2015-10-21 11:46:20,249 [INFO ] org.quartz.impl.StdSchedulerFactory :: Using default implementation for ThreadExecutor
2015-10-21 11:46:20,251 [INFO ] org.quartz.simpl.SimpleThreadPool :: Job execution threads will use class loader of thread: main
2015-10-21 11:46:20,263 [INFO ] org.quartz.core.SchedulerSignalerImpl :: Initialized Scheduler Signaller of type: class org.quartz.core.SchedulerSignalerImpl
2015-10-21 11:46:20,264 [INFO ] org.quartz.core.QuartzScheduler :: Quartz Scheduler v.2.1.7 created.
2015-10-21 11:46:20,265 [INFO ] org.quartz.simpl.RAMJobStore :: RAMJobStore initialized.
2015-10-21 11:46:20,266 [INFO ] org.quartz.core.QuartzScheduler :: Scheduler meta-data: Quartz Scheduler (v2.1.7) 'DefaultQuartzScheduler' with instanceId 'NON_CLUSTERED'
  Scheduler class: 'org.quartz.core.QuartzScheduler' - running locally.
  NOT STARTED.
  Currently in standby mode.
  Number of jobs executed: 0
  Using thread pool 'org.quartz.simpl.SimpleThreadPool' - with 10 threads.
  Using job-store 'org.quartz.simpl.RAMJobStore' - which does not support persistence. and is not clustered.

2015-10-21 11:46:20,266 [INFO ] org.quartz.impl.StdSchedulerFactory :: Quartz scheduler 'DefaultQuartzScheduler' initialized from default resource file in Quartz package: 'quartz.properties'
2015-10-21 11:46:20,266 [INFO ] org.quartz.impl.StdSchedulerFactory :: Quartz scheduler version: 2.1.7
2015-10-21 11:46:20,266 [INFO ] org.quartz.core.QuartzScheduler :: Scheduler DefaultQuartzScheduler_$_NON_CLUSTERED started.
2015-10-21 11:46:20,548 [INFO ] metabase.task :: 	loading tasks namespace:  metabase.task.sync-databases
2015-10-21 11:46:20,564 [INFO ] metabase.task.sync-databases :: Submitting sync-database task to scheduler
2015-10-21 11:46:20,577 [DEBUG] metabase.db.internal :: DB CALL:  Database * (where (limit (select* ENTITY19899) 1) {:is_sample true})
2015-10-21 11:46:20,749 [INFO ] metabase.driver.sync :: Syncing h2 database 'Sample Dataset'...
2015-10-21 11:46:20,755 [DEBUG] metabase.db.internal :: DB CALL:  Database * (where (limit (select* ENTITY24348) 1) {:id (events/object->model-id topic object)})
2015-10-21 11:46:20,767 [DEBUG] metabase.driver.sync :: Found new tables: #{"_METABASE_METADATA"}
2015-10-21 11:46:20,939 [DEBUG] metabase.driver.generic-sql.util :: Creating a new DB connection...
2015-10-21 11:46:23,081 [DEBUG] metabase.driver.sync :: [************······································] 😒   25% Synced table 'ORDERS'.
2015-10-21 11:46:23,107 [DEBUG] metabase.driver.sync :: [*************************·························] 😬   50% Synced table 'REVIEWS'.
2015-10-21 11:46:23,288 [DEBUG] metabase.driver.sync :: [*************************************·············] 😋   75% Synced table 'PRODUCTS'.
2015-10-21 11:46:24,347 [DEBUG] metabase.driver.sync :: [**************************************************] 😎  100% Synced table 'PEOPLE'.
2015-10-21 11:46:24,560 [INFO ] metabase.driver.sync :: Finished syncing h2 database Sample Dataset. (3813 ms)
2015-10-21 11:46:24,560 [INFO ] metabase.core :: Metabase Initialization COMPLETE
2015-10-21 11:46:24,561 [DEBUG] metabase.db.internal :: DB CALL:  Database * (where (limit (select* ENTITY24348) 1) {:id (events/object->model-id topic object)})
2015-10-21 11:46:24,590 [INFO ] metabase.core :: Launching Embedded Jetty Webserver with config:
 {:port 3000}

2015-10-21 11:46:24,637 [INFO ] org.eclipse.jetty.server.Server :: jetty-9.2.z-SNAPSHOT
2015-10-21 11:46:24,682 [INFO ] org.eclipse.jetty.server.ServerConnector :: Started ServerConnector@7af74333{HTTP/1.1}{0.0.0.0:3000}
2015-10-21 11:46:24,683 [INFO ] org.eclipse.jetty.server.Server :: Started @16452ms
2015-10-21 11:46:34,379 [DEBUG] metabase.db.internal :: DB CALL:  Setting [:value] (where (limit (select* ENTITY11921) 1) {:key (name k)})
2015-10-21 11:46:42,444 [DEBUG] metabase.db.internal :: DB CALL:  Setting [:value] (where (limit (select* ENTITY11921) 1) {:key (name k)})
```

**Comment by tlrobinson** [comment]
Actually it appears the connections are being blocked by the Mac app, so even if you add the database when running the jar directly it won't work once you restart the Mac app. We're working on a fix.

@dpaola2 This might be a different issue if you're running into problems when running the jar directly. Does the page load at all or just hang indefinitely? Any errors in the Web Inspector?

**Comment by dpaola2** [comment]
OK, I'll be patient :)

Just hangs indefinitely, no console errors: ![](https://www.dropbox.com/s/vowzp324boy2kb8/Screenshot%202015-10-21%2011.55.23.png?dl=1)

**Comment by tbeseda** [comment]
I was able to start the jar (thanks to Homebrew I could easily grab the JDK: `$ brew cask install java` -- I'm not a Java dev).
I was able to navigate to Metabase on port 3000 from my browser.
I ran through the setup again, I don't think it connected to the db the Mac app was using.
I was not able to add my Heroku db from the setup.
I was able to add my Heroku db from the admin panel after I skipped adding it in setup.

Looks like I have something to play with :smile: 

Thanks for the help @tlrobinson -- looking forward to updates
